swap/src/database/bob.rs

```diff
@@ -56,23 +56,23 @@ impl From<BobState> for Bob {
                 btc_amount,
                 change_address,
             },
-            BobState::SwapSetupCompleted(state2) => Bob::ExecutionSetupDone { state2 },
-            BobState::BtcLocked(state3) => Bob::BtcLocked { state3 },
+            BobState::SwapSetupCompleted(state2) => Bob::ExecutionSetupDone { state2: *state2 },
```

Comment:
If the `Box`s here are annoying, you can just get rid of them and allow the respective clippy lint that comes up.